src/main/java/games/strategy/net/nio/Encoder.java

```diff
@@ -90,4 +91,17 @@ private void write(final MessageHeader header, final ObjectOutputStream out, fin
     }
     out.reset();
   }
+
+  /**
+   * Provide access to the raw buffer.
+   */
+  private static final class ByteArrayOutputStream2 extends ByteArrayOutputStream {
```

Comment:
In this case:
Feel free  to self merge, if you think the current name scheme is ok.
If you want to change it though, go ahead, and I will comment on it later...